colcon-test/action.yaml

```diff
@@ -0,0 +1,119 @@
+name: colcon-test
+description: ""
+
+inputs:
+  rosdistro:
+    description: ""
+    required: true
+  target-packages:
+    description: ""
+    required: true
+  build-depends-repos:
+    description: ""
+    required: false
+  label:
+    description: ""
+    required: false
+    default: ".*" # Run all tests
+  token:
+    description: ""
+    required: false
+    default: ${{ github.token }}
+outputs:
+  coverage-reports:
+    description: ""
+    value: ${{ steps.test.outputs.coverage-reports }}
+
+runs:
+  using: composite
+  steps:
+    - name: Show target packages
+      run: |
+        echo "target packages: ${{ inputs.target-packages }}"
+      shell: bash
+
+    - name: Set up colcon-mixin
+      run: |
+        colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml || true
+        colcon mixin update default
+      shell: bash
+
+    - name: Install packages for coverage commands
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install lcov python3-pip
+        pip3 install colcon-lcov-result colcon-coveragepy-result
+      shell: bash
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != ''}}
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
+        mkdir dependency_ws
+        vcs import dependency_ws < ${{ inputs.build-depends-repos }}
+      shell: bash
+
+    - name: Run rosdep install
+      run: |
+        sudo apt-get -yqq update
+        rosdep update
+        DEBIAN_FRONTEND=noninteractive rosdep install -yqq --from-paths . --ignore-src --rosdistro ${{ inputs.rosdistro }}
+      shell: bash
+
+    - name: Restore build files from cache
+      id: restore-build-files
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./build
+          ./install
+        key: build-${{ github.sha }}
+
+    - name: Test
```

Comment:
Since this is a long step, we can consider splitting this.

Reply:
I modified here: ff6b0ee How about it?

Reply:
Thank you! Will take a look again.